src/sources/socket/mod.rs

```diff
@@ -229,23 +229,20 @@ impl SourceConfig for SocketConfig {
 
         let schema_definition = match &self.mode {
             Mode::Tcp(config) => {
-                let host_key_path = config
+                let legacy_host_key = config
                     .host_key()
                     .as_ref()
-                    .and_then(|x| parse_value_path(x).ok())
                     .map_or_else(
-                        || LegacyKey::InsertIfEmpty(owned_value_path!("host")),
-                        LegacyKey::InsertIfEmpty,
-                    );
+                        || parse_value_path(log_schema().host_key()).ok(),
+                        |k| k.path.clone(),
+                    )
+                    .map(LegacyKey::InsertIfEmpty);
```

Comment:
nitpick: Would this be less awkward if `log_schema().host_key()` returned a parsed path value? Maybe that's something we're already moving towards, not sure.